### PR TITLE
fix: /tracker/enrollments does not return attributes, events and relationships [DHIS2-12906]

### PR DESCRIPTION
**Pull request by lucaCambi77**
This is a new PR for https://dhis2.atlassian.net/browse/DHIS2-12906.
It continues https://github.com/dhis2/dhis2-core/pull/12304 that got stuck for some git merging problem.
It adds  **attributes, events, relationships** as fields to the /enrollments endpoint.
With this PR, now these attributes are handled by `EnrollmentParams` a nested `TrackedEntityInstanceParams` property.
Thus, now for the requests to `tracker/trackedEntities` we can use these attributes to modify the nested fields.
For example, if we want to exclude relationships of the enrollments we can do `tracker/trackedEntities?fields=relationships,enrollments[!relationships]`
There will be also a follow up to this PR to do the same for the events nested field

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/EnrollmentParams.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import lombok.Value;
+import lombok.With;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@With
+@Value
+public class EnrollmentParams
+{
+    public static final EnrollmentParams TRUE = new EnrollmentParams( true, true, true, true, false, false );
```
[the remark this answers is not in the document]
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/EnrollmentParams.java:39]: yes i'll find a better way to do that

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fieldsmapper/EnrollmentFieldsParamMapper.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.hisp.dhis.fieldfiltering.FieldPath;
+import org.hisp.dhis.fieldfiltering.FieldPreset;
+
+public class EnrollmentFieldsParamMapper
+{
+    private EnrollmentFieldsParamMapper()
+    {
+    }
+
+    public static TrackedEntityInstanceParams map( List<String> fields )
```
Comment (enricocolasante) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fieldsmapper/EnrollmentFieldsParamMapper.java:49]: I would expect a `EnrollmentParams` from the map method of `EnrollmentFieldsParamMapper`.
Just a suggestion but maybe the method name could indicate that a `TrackedEntityInstanceParams` is returned instead.

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/EnrollmentParams.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    public static final EnrollmentParams FALSE = new EnrollmentParams( false, false, false, false, false, false );
+
+    // Root is only relevant if it is a nested Param
+    private boolean includeRoot;
```
[the remark this answers is not in the document]
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/EnrollmentParams.java]: ok i made `EnrollmentParams` independent and created another class `TrackedEntityInstanceEnrollmentParams` to handle`EnrollmentParams` inside  `TrackedEntityInstanceParams`

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fieldsmapper/EnrollmentFieldsParamMapper.java (hunk 2 above)
Comment (lucaCambi77) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fieldsmapper/EnrollmentFieldsParamMapper.java]: yes same comment as @teleivo , i've made `EnrollmentParams` so now it should return that

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java [hunk 4]
```diff
@@ -384,6 +377,35 @@ public Enrollment getEnrollment( User user, ProgramInstance programInstance, Tra
             }
         }
 
+        if ( params.getEnrollmentParams().isIncludeAttributes() )
+        {
+            Set<TrackedEntityAttribute> readableAttributes = trackedEntityAttributeService
```
[the remark this answers is not in the document]
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java:382]: Yes, i agree but I'd say to see that later and make a refractory of the `getEnrollment` method. I'll make an issue later for that

**Review by teleivo — APPROVED** [review]

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**